assets/javascripts/discourse/initializers/chat-setup.js

```diff
@@ -124,6 +127,156 @@ export default {
           }
         });
       });
+      api.addSidebarSection(
+        (BaseCustomSidebarSection, BaseCustomSidebarSectionLink) => {
+          const SidebarChatSectionLink = class extends BaseCustomSidebarSectionLink {
+            constructor({ channel }) {
+              super(...arguments);
+              this.channel = channel;
+            }
+
+            get name() {
+              return this.channel.chatable_id;
+            }
+
+            get route() {
+              return "chat.channel";
+            }
+
+            get model() {
```

Comment:
Couldn't we just pass channel here ?

Reply:
Yeah, it feels weird. It is because the route required `channelId` and `channelTitle` 

https://github.com/discourse/discourse-chat/blob/main/assets/javascripts/discourse/chat-route-map.js#L5

And after model requires `chatChannel` to set active channel

https://github.com/discourse/discourse-chat/blob/main/assets/javascripts/discourse/routes/chat-channel.js#L41

Reply:
I see, I will soon be able to remove this afterModel